agent/agent.go

```diff
@@ -731,7 +731,7 @@ func (a *Agent) killProcs(ctx context.Context, tc *taskContext, ignoreTaskGroupC
 	}
 
 	if a.shouldKill(tc, ignoreTaskGroupCheck) {
-		if tc.task.ID != "" {
+		if tc.task.ID != "" && tc.taskConfig != nil {
```

Comment:
an alternative that occurs to me is that on the case where `tc.taskConfig == nil` we could call `KillSpawnedProcs` with the empty string as the workdir (since we check for the empty string [here](https://github.com/evergreen-ci/evergreen/blob/ef19ea1fa0e2ca32b375a46402a2b7bae0bb463f/agent/util/subtree_linux.go#L160-L162) and [here](https://github.com/evergreen-ci/evergreen/blob/ef19ea1fa0e2ca32b375a46402a2b7bae0bb463f/agent/util/subtree_darwin.go#L91-L93))

Reply:
We have multiple implementations of `getPIDsToKill` and would have to remember to correctly handle an empty working directory in all of them. On the other hand, the disadvantage of the solution I proposed is that we wouldn't run `killProcs` as completely as we would before.

I pushed a different solution. In my new code, we avoid running `killProcs` in the `runPostGroupCommands` function, since there's no reason to. Let me know if that makes sense.